samples/snippets/fsharp/tour.fs

```diff
@@ -0,0 +1,635 @@
+
+    let boolean3 = not boolean1 && (boolean2 || false)
+
+    printfn "The expression 'not boolean1 && (boolean2 || false)' is %A" boolean3
+
+
+
+// ---------------------------------------------------------------
+//         Strings
+// ---------------------------------------------------------------
+
+module StringManipulation = 
+
+    let string1 = "Hello"
+    let string2  = "world"
+
+    /// Use @ to create a verbatim string literal
+    let string3 = @"c:\Program Files\"
+
+    /// Using a triple-quote string literal
+    let string4 = """He said "hello world" after you did"""
+
+    let helloWorld = string1 + " " + string2 // concatenate the two strings with a space in between
+    printfn "%s" helloWorld
+
+    /// A string formed by taking the first 7 characters of one of the result strings
+    let substring = helloWorld.[0..6]
+    printfn "%s" substring
+
+
+
+// ---------------------------------------------------------------
+//         Tuples (ordered sets of values)
+// ---------------------------------------------------------------
+
+module Tuples = 
+
+    /// A simple tuple of integers
+    let tuple1 = (1, 2, 3)
+
+    /// A function that swaps the order of two values in a tuple. 
+    /// QuickInfo shows that the function is inferred to have a generic type.
+    let swapElems (a, b) = (b, a)
+
+    printfn "The result of swapping (1, 2) is %A" (swapElems (1,2))
+
+    /// A tuple consisting of an integer, a string, and a double-precision floating point number
+    let tuple2 = (1, "fred", 3.1415)
+
+    printfn "tuple1: %A    tuple2: %A" tuple1 tuple2
+
+    /// The origin, represented as a struct tuple
+    let origin = struct (0, 0)
+
+    /// Explicitly annotate the inputs as struct tuples.  This is not required.
+    let getDistance ((x1, x2): struct(int*int)) ((y1, y2): struct(int*int)) =
+        sqrt ((x2 - x1)**2 + (y1 - y1)**2)
+
+    printfn "The distance between the (0, 0) and (12, 13) is: %A" (getDistance origin struct (12,13))
+    
+
+
+// ---------------------------------------------------------------
+//         Lists and list processing
+// ---------------------------------------------------------------
+
+module Lists = 
+
+    let list1 = [ ]            /// an empty list
+
+    let list2 = [ 1; 2; 3 ]    /// list of 3 elements
+
+    let list3 = 42 :: list2    /// a new list with '42' added to the beginning
+
+    let numberList = [ 1 .. 1000 ]  /// list of integers from 1 to 1000
+
+    /// A list containing all the days of the year
+    let daysList = 
+        [ for month in 1 .. 12 do
+              for day in 1 .. System.DateTime.DaysInMonth(2012, month) do 
+                  yield System.DateTime(2012, month, day) ]
+
+    /// A list containing the tuples which are the coordinates of the black squares on a chess board.
+    let blackSquares = 
+        [ for i in 0 .. 7 do
+              for j in 0 .. 7 do 
+                  if (i+j) % 2 = 1 then 
+                      yield (i, j) ]
```

Comment:
This is a very imperative implementation, contrasting with what I read in the topic. 

Would a more declarative implementation be readable to F# beginners?